Owncloud iOs Client/Tabs/FileTab/FilesViewController.m

```diff
@@ -1532,48 +1532,49 @@ - (void) cancelGetThumbnailByCell:(UITableViewCell *) cell {
-    
-    UIFont *appFont = [UIFont fontWithName:@"HelveticaNeue" size:16];
-    
-    footerLabel.font = appFont;
-    footerLabel.textColor = [UIColor grayColor];
-    footerLabel.backgroundColor = [UIColor clearColor];
-    footerLabel.textAlignment = NSTextAlignmentCenter;
     
-    NSString *folders;
-    NSString *files;
-    NSString *footerText;
-    if (_numberOfFiles > 1) {
-        folders = [NSString stringWithFormat:@"%d %@", _numberOfFiles, NSLocalizedString(@"files", nil)];
-    } else if (_numberOfFiles == 1){
-        folders = [NSString stringWithFormat:@"%d %@", _numberOfFiles, NSLocalizedString(@"file", nil)];
-    } else {
-        folders = @"";
-    }
-    if (_numberOfFolders > 1) {
-        files = [NSString stringWithFormat:@"%d %@", _numberOfFolders, NSLocalizedString(@"folders", nil)];
-    } else if (_numberOfFolders == 1){
-        files = [NSString stringWithFormat:@"%d %@", _numberOfFolders, NSLocalizedString(@"folder", nil)];
-    } else {
-        files = @"";
-    }
-    if ([folders isEqualToString:@""]) {
-        footerText = files;
-    } else if ([files isEqualToString:@""]) {
-        footerText = folders;
-    } else {
-        footerText = [NSString stringWithFormat:@"%@, %@", folders, files];
-    }
-    footerLabel.text = footerText;
+    [self obtainTheQuantityOfFilesAndFolders];
     
     dispatch_async(dispatch_get_main_queue(), ^{
+        //Set the text of the footer section
+        UIView *footerView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, self.tableView.frame.size.width, 40 + self.tabBarController.tabBar.frame.size.height)];
+        footerView.backgroundColor = [UIColor clearColor];
+        
+        UILabel *footerLabel = [[UILabel alloc] initWithFrame:CGRectMake(0, 0, self.tableView.frame.size.width, 40)];
+        
+        UIFont *appFont = [UIFont fontWithName:@"HelveticaNeue" size:16];
+        
+        footerLabel.font = appFont;
+        footerLabel.textColor = [UIColor grayColor];
+        footerLabel.backgroundColor = [UIColor clearColor];
+        footerLabel.textAlignment = NSTextAlignmentCenter;
+        
+        NSString *folders;
+        NSString *files;
+        NSString *footerText;
+        if (_numberOfFiles > 1) {
+            folders = [NSString stringWithFormat:@"%d %@", _numberOfFiles, NSLocalizedString(@"files", nil)];
+        } else if (_numberOfFiles == 1){
+            folders = [NSString stringWithFormat:@"%d %@", _numberOfFiles, NSLocalizedString(@"file", nil)];
+        } else {
+            folders = @"";
+        }
+        if (_numberOfFolders > 1) {
+            files = [NSString stringWithFormat:@"%d %@", _numberOfFolders, NSLocalizedString(@"folders", nil)];
+        } else if (_numberOfFolders == 1){
+            files = [NSString stringWithFormat:@"%d %@", _numberOfFolders, NSLocalizedString(@"folder", nil)];
+        } else {
+            files = @"";
+        }
```

Comment:
hmm... really don't like. You could abstract this into a function. There is no need to duplicate this code.